src/commands/node.rs

```diff
@@ -123,201 +127,257 @@ impl Display for Arch {
     }
 }
 
-/// Struct implementation for the `Node` command.
-#[derive(Clone)]
-pub struct Node {}
+/// Manage node versions
+#[derive(Debug, Parser)]
+pub struct Node {
+    #[clap(subcommand)]
+    cmd: NodeCommand,
+}
 
-impl Node {
-    pub async fn download(args: &ArgMatches) -> Result<()> {
-        match args.subcommand() {
-            Some(("use", version)) => {
-                let v: String = String::from(version.value_of("version").unwrap());
-                use_node_version(v).await;
-            }
-            Some(("install", versions)) => {
-                let v: Vec<&str> = versions.values_of("versions").unwrap().collect();
-                download_node_version(v).await;
-            }
-            Some(("remove", versions)) => {
-                let v: Vec<&str> = versions.values_of("versions").unwrap().collect();
-                remove_node_version(v).await;
+#[async_trait]
+impl VoltCommand for Node {
+    async fn exec(self, config: VoltConfig) -> Result<()> {
+        match self.cmd {
+            NodeCommand::Use(x) => x.exec(config).await,
+            NodeCommand::Install(x) => x.exec(config).await,
+            NodeCommand::Remove(x) => x.exec(config).await,
+        }
+    }
+}
+
+#[derive(Debug, Subcommand)]
+pub enum NodeCommand {
+    Use(NodeUse),
+    Install(NodeInstall),
+    Remove(NodeRemove),
+}
+
+/// Switch current node version
+#[derive(Debug, Parser)]
+pub struct NodeUse {
+    /// Version to use
+    version: String,
+}
+
+#[async_trait]
+impl VoltCommand for NodeUse {
+    async fn exec(self, config: VoltConfig) -> Result<()> {
+        if PLATFORM == Os::Windows {
+            #[cfg(target_os = "windows")]
+            use_windows(version).await;
```

Comment:
```suggestion
            use_windows(self.version).await;
```
This needs to be `self.version`; as is it gives this output on Windows:
```
❯ cargo build
error[E0425]: cannot find value `version` in this scope
   --> src\commands\node.rs:167:25
    |
167 |             use_windows(version).await;
    |                         ^^^^^^^
    |
help: you might have meant to use the available field
    |
167 |             use_windows(self.version).await;
    |                         ~~~~~~~~~~~~
help: consider importing this function
    |
19  | use isahc::version;
    |

For more information about this error, try `rustc --explain E0425`.
error: could not compile `volt` due to previous error
```

Reply:
Ahh, I am on linux so i missed it. Thanks for catching it.